scripts/rook.sh

```diff
@@ -1,10 +1,12 @@
 #!/bin/bash -E
 
-ROOK_VERSION=${ROOK_VERSION:-"v1.2.7"}
+ROOK_VERSION=${ROOK_VERSION:-"v1.4.9"}
 ROOK_DEPLOY_TIMEOUT=${ROOK_DEPLOY_TIMEOUT:-300}
 ROOK_URL="https://raw.githubusercontent.com/rook/rook/${ROOK_VERSION}/cluster/examples/kubernetes/ceph"
 ROOK_BLOCK_POOL_NAME=${ROOK_BLOCK_POOL_NAME:-"newrbdpool"}
-ROOK_CEPH_CLUSTER_VERSION="v14.2.10"
+BUILD_ENV="build.env"
+CEPH_IMAGE=$(awk -F : '/^BASE_IMAGE=/ {print $NF}' "${BUILD_ENV}")
+ROOK_CEPH_CLUSTER_VERSION=${ROOK_CEPH_CLUSTER_VERSION:-${CEPH_IMAGE}}
```

Comment:
`ROOK_CEPH_CLUSTER_VERSION_IMAGE_PATH` below already contains `ceph/ceph:` before placing the version. you can make things simpler by just using the `BASE_IMAGE` there in case `ROOK_CEPH_CLUSTER_VERSION` is not set.

Add `ROOK_CEPH_CLUSTER_VERSION` commented out in `build.env`.

Reply:
But then I need to change the BASE_IMAGE version now because it is needed to run snapshot retention feature. Is that ok?

Reply:
I feel that will add more complexity

Reply:
What I do not like here:

1. reading from `build.env`, assuming it is in the current working directory
2. using `build.env` eventhough something else might set a different value for `BASE_IMAGE`
3. images do not need a `:` to separate `name:version`, it is optional (defaults to `:latest`)
4. assuming the ceph image always starts with `ceph/ceph`

I'd much more prefer using the following logic:

- introduce `ROOK_CEPH_CLUSTER_IMAGE`
- if `ROOK_CEPH_CLUSTER_IMAGE` is set, do not change the value
- if `ROOK_CEPH_CLUSTER_IMAGE` is not set, use `BASE_IMAGE`
- if `BASE_IMAGE` is not set default `ROOK_CEPH_CLUSTER_IMAGE` to `ceph/ceph:v14.2.10`

That way, you can always use `ROOK_CEPH_CLUSTER_IMAGE` in the function below. It also offers a way to use a patched ceph release from a different registry for specific testing.

Reply:
the above looks good,  why use BASE_IMAGE? if the `ROOK_CEPH_CLUSTER_IMAGE` is not set, install what comes default in rook itself.

Reply:
When we use `BASE_IMAGE`, we know that client and server are compatible. Using the Rook default should be fine too. I'm not sure which would be better, so either is fine with me.